### PR TITLE
`App Service` - Add authV2 to Web Apps

### PR DESCRIPTION
**Pull request by jackofallops**
* `azurerm_linux_web_app` - add support for AuthV2 (EasyAuthV2) `auth_settings_v2`
* `azurerm_linux_web_app_slot` - add support for AuthV2 (EasyAuthV2) `auth_settings_v2`
* `azurerm_windows_web_app` - add support for AuthV2 (EasyAuthV2) `auth_settings_v2`
* `azurerm_windows_web_app_slot` - add support for AuthV2 (EasyAuthV2) `auth_settings_v2`
* **Data Source:** `azurerm_linux_web_app` - add support for AuthV2 (EasyAuthV2) `auth_settings_v2`
* **Data Source:**`azurerm_windows_web_app` - add support for AuthV2 (EasyAuthV2) `auth_settings_v2`

Partially addresses #12928 (Function Apps to follow)

### CODE REVIEW

File: internal/services/appservice/windows_web_app_slot_resource.go [hunk 1]
```diff
@@ -746,13 +765,37 @@ func (r WindowsWebAppSlotResource) Update() sdk.ResourceFunc {
...
+						FacebookAppSecret:                 pointer.To(""),
+						GitHubClientSecret:                pointer.To(""),
+						TwitterConsumerSecret:             pointer.To(""),
+						MicrosoftAccountClientSecret:      pointer.To(""),
+					}
+					updateLogs = true
+				}
 				if _, err := client.UpdateAuthSettingsSlot(ctx, id.ResourceGroup, id.SiteName, *authUpdate, id.SlotName); err != nil {
-					return fmt.Errorf("updating Auth Settings for Windows %s: %+v", id, err)
+					return fmt.Errorf("updating Auth Settings for Linux %s: %+v", id, err)
```
Comment (manicminer) [inline internal/services/appservice/windows_web_app_slot_resource.go:787]:
```suggestion
					return fmt.Errorf("updating Auth Settings for Windows %s: %+v", id, err)
```

File: internal/services/appservice/windows_web_app_resource.go [hunk 2]
```diff
@@ -800,13 +820,37 @@ func (r WindowsWebAppResource) Update() sdk.ResourceFunc {
...
+						FacebookAppSecret:                 pointer.To(""),
+						GitHubClientSecret:                pointer.To(""),
+						TwitterConsumerSecret:             pointer.To(""),
+						MicrosoftAccountClientSecret:      pointer.To(""),
+					}
+					updateLogs = true
+				}
 				if _, err := client.UpdateAuthSettings(ctx, id.ResourceGroup, id.SiteName, *authUpdate); err != nil {
-					return fmt.Errorf("updating Auth Settings for Windows %s: %+v", id, err)
+					return fmt.Errorf("updating Auth Settings for Linux %s: %+v", id, err)
```
Comment (manicminer) [inline internal/services/appservice/windows_web_app_resource.go:842]:
```suggestion
					return fmt.Errorf("updating Auth Settings for Windows %s: %+v", id, err)
```

File: internal/services/appservice/windows_web_app_resource.go [hunk 3]
```diff
@@ -473,6 +484,14 @@ func (r WindowsWebAppResource) Read() sdk.ResourceFunc {
 				return fmt.Errorf("reading Auth Settings for Windows %s: %+v", id, err)
 			}
 
+			var authV2 web.SiteAuthSettingsV2
+			if pointer.From(auth.ConfigVersion) == "v2" {
+				authV2, err = client.GetAuthSettingsV2(ctx, id.ResourceGroup, id.SiteName)
+				if err != nil {
+					return fmt.Errorf("reading authV2 settings for Linux %s: %+v", *id, err)
```
Comment (manicminer) [inline internal/services/appservice/windows_web_app_resource.go:491]:
```suggestion
					return fmt.Errorf("reading authV2 settings for Windows %s: %+v", *id, err)
```

File: internal/services/appservice/windows_web_app_resource.go [hunk 4]
```diff
@@ -392,6 +396,13 @@ func (r WindowsWebAppResource) Create() sdk.ResourceFunc {
 				}
 			}
 
+			authv2 := helpers.ExpandAuthV2Settings(webApp.AuthV2Settings)
+			if authv2.SiteAuthSettingsV2Properties != nil {
+				if _, err = client.UpdateAuthSettingsV2(ctx, id.ResourceGroup, id.SiteName, *authv2); err != nil {
+					return fmt.Errorf("updating AuthV2 settings for Linux %s: %+v", id, err)
```
Comment (manicminer) [inline internal/services/appservice/windows_web_app_resource.go:402]:
```suggestion
					return fmt.Errorf("updating AuthV2 settings for Windows %s: %+v", id, err)
```

**Review by manicminer — APPROVED** [review]
Couple of minor errtext things, otherwise LGTM 🏆

**Review by mbfrahry — APPROVED** [review]
LGTM

**Pull request merged**